pyproject.toml

```diff
@@ -1,118 +1,130 @@
-"Homepage" = "https://truss.baseten.co"
+[project.urls]
+Homepage = "https://truss.baseten.co"
 "Bug Reports" = "https://github.com/basetenlabs/truss/issues"
-"Documentation" = "https://truss.baseten.co"
-"Baseten" = "https://baseten.co"
-
-[tool.poetry.dependencies]
-blake3 = "^0.3.3"
-boto3 = "^1.26.157"
-fastapi = ">=0.109.1"
-google-cloud-storage = "2.10.0"
-httpx = "^0.24.1"
-huggingface_hub = ">=0.19.4"
-inquirerpy = "^0.3.4"
-Jinja2 = "^3.1.2"
-loguru = ">=0.7.2"
-msgpack = ">=1.0.2"
-msgpack-numpy = ">=0.4.8"
-numpy = ">=1.23.5"
-packaging = ">=20.9"
-pathspec = ">=0.9.0"
-psutil = ">=5.9.4"
-pydantic = ">=1.10.0"
-python = ">=3.8,<3.12"
-python-json-logger = ">=2.0.2"
-python-on-whales = "^0.68.0"
-PyYAML = ">=6.0"
-rich = "^13.4.2"
-rich-click = "^1.6.1"
-single-source = "^0.3.0"
-tenacity = "^8.0.1"
-watchfiles = "^0.19.0"
-
-[tool.poetry.group.builder.dependencies]
-blake3 = "^0.3.3"
-boto3 = "^1.26.157"
-click = "^8.0.3"
-fastapi = ">=0.109.1"
-google-cloud-storage = "2.10.0"
-httpx = "^0.24.1"
-huggingface_hub = ">=0.19.4"
-Jinja2 = "^3.1.2"
-loguru = ">=0.7.2"
-packaging = ">=20.9"
-pathspec = ">=0.9.0"
-psutil = ">=5.9.4"
-python = ">=3.8,<3.12"
-python-json-logger = ">=2.0.2"
-PyYAML = ">=6.0"
-requests = ">=2.31"
-single-source = "^0.3.0"
-tenacity = "^8.0.1"
-uvicorn = "^0.24.0"
+Documentation = "https://truss.baseten.co"
+Baseten = "https://baseten.co"
+repository = "https://github.com/basetenlabs/truss"
 
-[tool.poetry.dev-dependencies]
-black = "^22.6.0"
-coverage = "^6.4.1"
-dockerfile = "^3.2.0"
-flake8 = "^4.0.1"
-ipdb = "^0.13.9"
-ipykernel = "^6.16.0"
-ipython = "^7.16"
-isort = "^5.10.1"
-nbconvert = "^7.2.1"
-pre-commit = "^2.18.1"
-pytest = "7.2.0"
-pytest-cov = "^3.0.0"
-types-PyYAML = "^6.0.12.12"
-types-setuptools = "^69.0.0.0"
+[project.scripts]
+truss = "truss.cli:truss_cli"
 
-[tool.poetry.scripts]
-truss = 'truss.cli:truss_cli'
+[tool.rye]
+managed = true
+dev-dependencies = [
+    "flask<3.0.0,>=2.3.3",
+    "httpx[cli]<1.0.0,>=0.24.1",
+    "mypy<2.0.0,>=1.0.0",
+    "pytest-split<1.0.0,>=0.8.1",
+    "requests-mock>=1.11.0",
+    "types-requests>=2.31.0.2",
+    "uvicorn>=0.24.0",
+    "uvloop>=0.17.0",
+    "flake8>=7.0.0",
+]
 
-[tool.poetry.group.dev.dependencies]
-flask = "^2.3.3"
-httpx = { extras = ["cli"], version = "^0.24.1" }
-mypy = "^1.0.0"
-pytest-split = "^0.8.1"
-requests-mock = ">=1.11.0"
-types-requests = ">=2.31.0.2"
-uvicorn = ">=0.24.0"
-uvloop = ">=0.17.0"
+## Package Build
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
 
+[tool.hatch.version]
+path = "truss/__init__.py"
 
-[tool.poetry.group.truss-chains.dependencies]
-datamodel-code-generator = "^0.25.4"
-libcst = "<1.2.0"
-autoflake = "<=2.2"
+[tool.hatch.metadata]
+allow-direct-references = true
 
-[build-system]
-build-backend = "poetry.core.masonry.api"
-requires = ["poetry-core>=1.2.1"]
+[tool.hatch.build.targets.wheel]
+only-packages = true
+packages = ["src/truss"]
 
+## Lint Tools
```

Comment:
Any chance we can bump our lower python limit from `py3.8` to (at least) `py3.9`?

Reply:
sure. will add